src/UI/Tables/Stateable.elm

```diff
@@ -0,0 +1,746 @@
+
+-- Filters
+
+
+type alias Filters msg item columns =
+    Filters.Filters msg item columns
+
+
+type alias Strategy msgs value item =
+    Filters.Strategy msgs value item
+
+
+withFilters : Filters msg item columns -> StateableTable msg item columns -> StateableTable msg item columns
+withFilters filters (Table prop opt) =
+    Table prop { opt | filters = Just filters }
+
+
+filtersEmpty : Filters msg item T.Zero
+filtersEmpty =
+    NArray.empty
+
+
+filtersPush : Filters.Filter msg item -> Filters msg item columns -> Filters msg item (T.Increase columns)
+filtersPush filter accu =
+    NArray.push filter accu
+
+
+localSingleTextFilter :
+    Maybe String
+    -> (item -> String)
+    -> Filters msg item columns
+    -> Filters msg item (T.Increase columns)
+localSingleTextFilter initValue get accu =
+    let
+        applier item current =
+            get item
+                |> String.toLower
+                |> String.contains (String.toLower current)
+    in
+    { initial = initValue, strategy = filterLocal applier }
+        |> Filters.SingleTextFilter
+        |> swap filtersPush accu
+
+
+filterLocal : (item -> value -> Bool) -> Strategy msgs value item
+filterLocal isKept =
+    Filters.Local isKept
+
+
+remoteSingleTextFilter :
+    Maybe String
+    -> (String -> msg)
+    -> Filters msg item columns
+    -> Filters msg item (T.Increase columns)
+remoteSingleTextFilter initValue editMsg accu =
+    let
+        msgs =
+            { editMsg = editMsg }
+    in
+    { initial = initValue, strategy = filterRemote msgs }
+        |> Filters.SingleTextFilter
+        |> swap filtersPush accu
+
+
+filterRemote : msgs -> Strategy msgs value item
+filterRemote msgs =
+    Filters.Remote msgs
+
+
+
+-- Width
+
+
+{-| Applies [`Element.width`](/packages/mdgriffith/elm-ui/latest/Element#width) to the component.
+
+    Table.withWidth
+        (Element.fill |> Element.minimum 220)
+        someTable
+
+-}
+withWidth : Element.Length -> StateableTable msg item columns -> StateableTable msg item columns
+withWidth width (Table prop opt_) =
+    Table prop { opt_ | width = width }
+
+
+
+-- Rendering
+
+
+{-| End of the builder's life.
+The result of this function is a ready-to-insert Elm UI's Element.
+-}
+renderElement : RenderConfig -> StateableTable msg item columns -> Element msg
+renderElement renderConfig (Table prop opt) =
+    case opt.responsive of
+        Just responsive ->
+            if RenderConfig.isMobile renderConfig then
+                mobileView renderConfig prop opt responsive
+
+            else
+                desktopView renderConfig prop opt
+
+        Nothing ->
+            desktopView renderConfig prop opt
+
+
+
+-- Mobile rendering
+
+
+mobileView :
+    RenderConfig
+    -> StateableConfig msg item columns
+    -> Options msg item columns
+    -> Responsive msg item columns
+    -> Element msg
+mobileView renderConfig prop opt responsive =
+    ListView.toggleableList
+        { detailsShowLabel = "Expand"
```

Comment:
Not related with this pr, but having these Strings here is a problem regarding i18n